python/lsst/ts/standardscripts/base_close_loop.py

```diff
@@ -0,0 +1,492 @@
+            type: object
+            properties:
+              mode:
+                description: >-
+                    Mode to use for the script. If set to "cwfs", the script will
+                    use the corner wavefront sensors, if set to "fam" the script
+                    will use the full array mode.
+                type: string
+                enum: ["cwfs", "fam"]
+                default: cwfs
+              filter:
+                description: Which filter to use when taking intra/extra focal images.
+                type: string
+                default: empty_1
+              exposure_time:
+                description: The exposure time to use when taking images (sec).
+                type: number
+                default: 30.
+              dz:
+                description: De-focus to apply when acquiring the intra/extra focal images (mm).
+                type: number
+                default: 0.8
+              threshold:
+                description: >-
+                    DOF threshold for convergence (um). If DOF offsets are
+                    smaller than this value, the script will stop.
+                type: array
+                items:
+                  type: number
+                  minimum: 0
+                minItems: 50
+                maxItems: 50
+                default: [0.004, 0.004, 0.004, 0.004, 0.004,
+                          0.004, 0.004, 0.004, 0.004, 0.004,
+                          0.004, 0.004, 0.004, 0.004, 0.004,
+                          0.004, 0.004, 0.004, 0.004, 0.004,
+                          0.004, 0.004, 0.004, 0.004, 0.004,
+                          0.004, 0.004, 0.004, 0.004, 0.004,
+                          0.004, 0.004, 0.004, 0.004, 0.004,
+                          0.004, 0.004, 0.004, 0.004, 0.004,
+                          0.004, 0.004, 0.004, 0.004, 0.004,
+                          0.004, 0.004, 0.004, 0.004, 0.004]
+              max_iter:
+                description: Maximum number of iterations.
+                Note, if apply_corrections is False, the script
+                will take [max_iter] pairs of images.
+                type: integer
+                default: 5
+              program:
+                description: >-
+                    Optional name of the program this dataset belongs to.
+                type: string
+                default: CWFS
+              reason:
+                description: Optional reason for taking the data.
+                anyOf:
+                  - type: string
+                  - type: "null"
+                default: null
+              used_dofs:
+                oneOf:
+                  - type: array
+                    items:
+                      type: integer
+                      minimum: 0
+                      maximum: 49
+                  - type: array
+                    items:
+                      type: string
+                      enum: [
```

Comment:
I think you could dynamically generate this by doing something like:

```
enum: {[dof_name.name for dof_name in DOFName]}
```

and covert this to an f-string, by adding a "f" at the top, e.g.:

```
schema_yaml = f"""
...          
```